v4/export/writer_util.go

```diff
@@ -182,6 +182,86 @@ func WriteInsert(tblIR TableDataIR, w io.Writer) error {
 	return wp.Error()
 }
 
+func WriteInsertInCsv(tblIR TableDataIR, w io.Writer, noHeader bool) error {
+	fileRowIter := tblIR.Rows()
+	if !fileRowIter.HasNext() {
+		return nil
+	}
+
+	bf := pool.Get().(*bytes.Buffer)
+	if bfCap := bf.Cap(); bfCap < lengthLimit {
+		bf.Grow(lengthLimit - bfCap)
+	}
+
+	wp := newWriterPipe(w)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		wp.Run(ctx)
+		wg.Done()
+	}()
+	defer func() {
+		cancel()
+		wg.Wait()
+	}()
+
+	var (
+		csvHeader bytes.Buffer
+		row       = MakeRowReceiver(tblIR.ColumnTypes())
+		counter   = 0
+		err       error
+	)
+
+	if !noHeader || len(tblIR.ColumnNames()) == 0 {
+		for i, col := range tblIR.ColumnNames() {
+			csvHeader.WriteString(col)
+			if i != len(tblIR.ColumnTypes())-1 {
+				csvHeader.WriteString(",")
+			}
+		}
+		csvHeader.WriteString("\n")
+	}
+
+	bf.WriteString(csvHeader.String())
```

Comment:
```suggestion
	if !noHeader || len(tblIR.ColumnNames()) == 0 {
		for i, col := range tblIR.ColumnNames() {
			bf.WriteByte(doubleQuotationMark)
			escape(s.RawBytes, bf, escapeBackslash)
			bf.WriteByte(doubleQuotationMark) 
			if i != len(tblIR.ColumnTypes())-1 {
				bf.WriteByte(',')
			}
		}
		bf.WriteByte('\n')
	}
```

1. we can directly write to `bf`.
2. column name *can* contain characters that need to be escaped.